CN0561 hdl update

laurent-19 opened the pull request:
## Pull Request Description

This update is based on the changes made to the HDL SPIE architecture. 
The trigger for the SPIE offload is generated using the PWM ip for achieving higher speed for the data transfer. 
ADC filtering had to be updated to allow for ~ 1 MHz output data rates. 
Register dump was added for checking the new settings and general debugging.

The new NO-OS was tested on hardware using the current .xsa for the cn0561 project.

Might require a guideline check.

► CN0561 new HDL Reference Design
https://wiki.analog.com/resources/eval/user-guides/cn0561/hdl
► AD4134 ADC Data Sheet
https://www.analog.com/media/en/technical-documentation/data-sheets/ad4134.pdf

## PR Type
[x] New feature (change that adds new functionality)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=1910) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/analogdevicesinc/no-OS?pullRequest=1910) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/analogdevicesinc/no-OS?pullRequest=1910) it.</sub>

amiclaus commented on projects/cn0561/src/cn0561.c:
```diff
@@ -115,6 +118,7 @@ int main()
 		.platform_ops = &xil_gpio_ops,
 		.extra = &gpio_extra_param
 	};
+	MAX_SPEED_HZ = 48000000;
```
this looks like a re-initialization? is the previous value 24000000 used anywhere? Also for variables use lowercase letters.

laurent-19 replied:
it should be used when using the cora carrier board and was the default in the original project.

amiclaus replied:
ack. but please address the second part of the review (lowercase)

amiclaus commented on projects/cn0561/src/cn0561.c:
```diff
@@ -65,6 +65,8 @@
 #include "clk_axi_clkgen.h"
 #include "axi_dmac.h"
 
+#define CN0561_ZED_CARRIER
```
please remove. this is handled in the builds.json. Looks like you added it only for testing locally.

amiclaus commented on projects/cn0561/src/cn0561.c:
```diff
@@ -217,6 +217,12 @@ int main()
 	if (ret != 0)
 		return -1;
 
+	for(int adc_channel = CH0; adc_channel <= CH3; adc_channel++) {
```
nit: `for (` . also declaration of `adc_channel` should be placed at the start of the function.

amiclaus reviewed (commented):
please sign the CLA.

laurent-19 commented on projects/cn0561/src/cn0561.c:
```diff
@@ -331,6 +331,29 @@ int main()
 	}
 
+	uint8_t reg_data;
+	uint8_t reg_addr;
```
I would say is better to have the reg dump there, but indeed, some guideline updates are needed. 
Agree on keeping the reg dump with the changes or just exclude it?

amiclaus replied:
hmm. i'd not provide this by default. you can add a define here commented for CN0561_REG_DUMP. https://github.com/analogdevicesinc/no-OS/blob/c87d5f65fe63e6b1b1f2ed2595bee130cb258bc0/projects/cn0561/src/parameters.h#L77 . And if someone wants to use it then fine. Let's see also what others say on this topic also.

laurent-19 replied:
Yes, a define sounds fair.

amiclaus commented on drivers/axi_core/axi_pwmgen/axi_pwm.c:
```diff
@@ -307,6 +307,7 @@ int32_t axi_pwm_init(struct no_os_pwm_desc **desc,
 	pwm_desc->duty_cycle_ns = param->duty_cycle_ns;
 	pwm_desc->period_ns = param->period_ns;
 	pwm_desc->polarity = param->polarity;
+	pwm_desc->phase_ns = param->phase_ns;
```
Actually we need this. See line 349.

buha replied:
so this commit is good but please edit the commit description as @amiclaus said to look like this:

```
axi_pwm: use phase_ns initialization parameter (or something like that)

...
description
...

Signed-off-by: laurent-19 <laurentiu.popa@analog.com>
```

amiclaus reviewed (commented):
please do not mark as resolved comments as long as there is no new version of the patches in the PR.

amiclaus reviewed (commented):
do not add extra commit for review implementations. just amend each change to the appropriate existing patch. also add a changelog mentioning what you changed/added with your last force push.

laurent-19 commented:
Updated commits according to guidelines and discussion:
* defined data clocks for different carriers
* kept register dump for ad4134 device
* kept phase shift initialization in ad4134 driver file

buha commented on projects/cn0561/src/cn0561.c:
```diff
@@ -330,6 +334,29 @@ int main()
 		}
 	}
 
+#ifdef CN0561_REG_DUMP
```
this is good but i think it'd be more useful if you'd put it in a function so that it could be used from anywhere at run-time, not just this one place

laurent-19 replied:
Should I add this in the ad713x driver then?

buha replied:
My opinion is that something like that is useful in the driver, others may differ... I think there are some no-OS drivers which have a regmap dump API.

buha replied:
in most cases it's a simple for loop with a register read call, so no dedicated function is needed

but such a register dump function it's useful when there are gaps (like for this device) in the regmap

laurent-19 commented:
Updated commit history:
Added full description to axi_pwm commit.
Moved spi_reg_dump print function definition in the ad713x driver and used it in the cn0561 project.

buha commented:
please run `ci/astyle.sh` and recommit
